### ISSUE TITLE
Option to deselect the `mark this order complete` checkbox by default

### ISSUE THREAD

**Issue by madeincosmos** [Automattic/woocommerce-services#2005]
User in 2883635-zen is complaining about the **Mark this order as complete and notify the customer** checkbox showing now when buying a label:

![Screenshot](https://d.pr/i/iUS6h3+)
Screenshot: https://d.pr/i/iUS6h3

They say this goes against their workflow and creates a lot of confusion with their customers if they forget to uncheck it.

I'm asking them for more details about what their workflow is to understand why this is a problem.

In the meantime, how complicated it would be to add an option in WooCommerce Services settings to leave this box unchecked by default?

**Comment by c-shultz** [comment]
Thanks for bringing this up and creating an issue! Do you think this would be a common issue for merchants? Another approach would be for us to remember if they've unchecked it and leave it unchecked next time around.

/cc @elizaan36

**Comment by DustinHartzler** [comment]
I have a chatter right now with the same suggestion: 12055358-hc (have an option to leave unchecked by default).

They mention that they pre-print the labels and shipping is done on current days, then they change all the Completed when they are actually shipped.

**Comment by jacoswan** [comment]
To echo the reply above, it's exactly the same what @madeincosmos' user (2883635-zen) replied with to say their workflow is: 

> I may print the label several days in advance of the normally scheduled USPS pickup. I only send out shipping notifications and close the order after USPS has picked up the shipments and the order has actually shipped.

**Comment by elizaan36** [comment]
We had a conversation on this in Slack ages ago and then I meant to comment on here. Wanted to say that we should remember the users last choice as @c-shultz suggested, as it will likely be their preference going forward.

In the future we could consider including an option in settings to either mark the order as complete or not when printing labels.

**Comment by LuckyHerbals** [comment]
I am trying to default to mark order complete and notify the customer when postage is purchased/printed. How can this be resolved? We really liked that feature before the latest update..

**Comment by madeincosmos** [comment]
User in 3476170-zen would also prefer to have it checked by default. Would be fantastic if the plugin remembered previous user's preferences.

**Comment by madeincosmos** [comment]
One more user who prefers to have it checked by default in 11067239-zen

**Comment by waclawjacek** [comment]
Is this something that should be stored on a per-user or site-wide basis?

**Paper size** is another similar setting that is currently updated (at least) when:

* Its value is changed manually via `WooCommerce > Settings > Shipping > WooCommerce Shipping`.
* A label is printed.

The "Paper size" setting is stored site-wide, i.e. the next label purchase's paper size will default to the size selected during the last print request, regardless of who requested it.

Should the **Mark this order as complete and notify the customer** checkbox's value reflect this behavior for consistency?

Saving it per-user would have the benefit of accommodating different user workflows within a single site. This could be useful if one administrator prefers to have orders marked as complete automatically, and another administrator prefers to do this manually later. 

If the workflow is the same for all users, storing it per-user would require each administrator to change this setting on their end, potentially causing more trouble than it's worth.

And finally, should the value stored for the **Mark this order as complete and notify the customer** checkbox also apply to the **Notify the customer with shipment details** checkbox which can be seen when printing labels for an order that is already completed?

**Comment by elizaan36** [comment]
Great questions @waclawjacek 

It should be saved site-wide rather than per user, consistent with the paper selection as you pointed out. This will solve the majority of issues we've seen here.

And saving last selected for the other checkbox (when an order is complete) makes sense too.👍

**Comment by waclawjacek** [comment]
Should the value for the **Notify the customer with shipment details** checkbox be stored as a separate setting?

**Comment by elizaan36** [comment]
Yeah, I think it should be stored as a separate setting. It's a different use case altogether since the order has already been fulfilled. It's probably a one-off event, for example reprinting a label in which case the merchant may not want the checkbox to be selected by default (even if their unfulfilled orders have the checkbox selected by default.)

**Comment by Abdalsalaam** [comment]
Shipping related work now happens in WooCommerce Shipping and we recommend using it: https://wordpress.org/plugins/woocommerce-shipping/
